regression/cbmc-java-concurrency/several_threads/A.java

```diff
@@ -0,0 +1,97 @@
+import java.lang.Thread;
+import org.cprover.CProver;
+
+public class A
+{
+
+  // expected verification success
+  public void me()
+  {
+    B b= new B();
+    b.me();
+  }
+
+  // expected verification failure
+  public void me2()
+  {
+    C c= new C();
+    c.me();
+  }
+
+  // Create 32 Threads, no shared variables
+  // expected verification success
+  public void me3()
+  {
+    for(int i = 0; i<8; i++)
+    {
+       D d = new D();
+       d.start();
+    }
+  }
+}
+
+class D extends Thread
+{
+  int x = 0;
+
+  @Override
+  public void run()
+  {
+    x = 44;
+    int local_x = x;
+    assert(local_x == 44 || local_x == 10);
+  }
+  public void setX()
+  {
+    x=10;
+  }
+}
+
+
+
+
+class B implements Runnable
+{
+  int x = 0;
```

Comment:
Add comment:
```java
// FIXME: the assertion below does not always hold as per the Java spec (because x is not volatile), but cbmc currently doesn't reason about volatile variables
```